docs/source/tensor_view.rst

```diff
@@ -0,0 +1,71 @@
+.. currentmodule:: torch
+
+.. _tensor-view-doc:
+
+Tensor Views
+=================
+
+PyTorch allows a tensor to be `View` of an existing tensor. View tensor shares the same underlying storage
+with its base tensor. Supporting `View` allows us to do fast and memory efficient
+reshaping, slicing and element-wise operations.
+
+For example, to get a view of an existing tensor `t`, you can call `t.view(...)`.
+
+::
+
+    >>> t = torch.rand(4, 4)
+    >>> b = t.view(2, 8)
+    >>> t.storage().data_ptr() == b.storage().data_ptr()  # `t` and `b` share the same storage.
+    True
+    # Modifying views will change the original tensor as well.
+    >>> b[0][0] = 3.14
+    >>> t[0][0]
+    tensor(3.14)
+
+Note that since views share underlying storage with the original tensor, if you edit the data
+in the view, it will be reflected in the base tensor as well.
+
+Other than :meth:`~torch.Tensor.view` shown above, there're more PyTorch ops taking advantange of efficiency of views.
+For reference, here’s a list of all ops in PyTorch that return view of input tensors:
+
+- Basic indexing return views.
+
+  When accessing the contents of a tensor via indexing, PyTorch follows Numpy behaviors
+  that basic indexing returns views, while advanced indexing returns a copy.
+  Assignment with both basic and advanced indexing is in-place. See more examples in
+  `Numpy indexing documentation <https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html>`_.
+- :meth:`~torch.Tensor.as_strided`
+- :meth:`~torch.Tensor.diagonal`
+- :meth:`~torch.Tensor.expand`
+- :meth:`~torch.Tensor.narrow`
+- :meth:`~torch.Tensor.permute`
+- :meth:`~torch.Tensor.select`
+- :meth:`~torch.Tensor.squeeze`
+- :meth:`~torch.Tensor.transpose`
```

Comment:
Transpose is a really common example where users mistakenly think data movement occurs when transpose is called, but nothing actually happens except for a view being taken until .contiguous() is called. This is important to understand for performance, too.